src/stream.iced

```diff
@@ -0,0 +1,26 @@
+stream = require('stream')
+enc = require('./encoding')
+
+desired_block_size = 4096
+calculate_block_size = (input_length) ->
+  input_length*Math.floor(desired_block_size/input_length)
+
+exports.StreamEncoder = class StreamEncoder extends stream.Transform
+
+  constructor : (@encoder) ->
+    super({highWaterMark : calculate_block_size(encoder.in_block_len)})
+
+  _transform : (chunk, encoding, cb) ->
+    @push(@encoder.encode(chunk))
+    @read(0)
```

Comment:
The whole stream is placed into a buffer for easy comparison with T.equals - running T.equals on individual chunks would be difficult, but I could do it if necessary.